### PR TITLE
Feat: change remux after initial creation

### PR DESCRIPTION
**Pull request by brandonocasey**
For fmp4 playlists with ts segment discontinuities the track ids of the fmp4 files and the ts segments that we transmux to fmp4 must match up.

### CODE REVIEW

File: lib/partial/transmuxer.js [hunk 1]
```diff
@@ -352,6 +352,15 @@ var Transmuxer = function(options) {
     }
   };
 
+  this.setRemux = function(val) {
```
Comment (brandonocasey) [inline lib/partial/transmuxer.js]: currently this won't do anything as we don't have a `coalesceStream` for the partial transmuxer. This also means fmp4 and ts content will not work with the partial transmuxer right now.

Comment (ldayananda) [reply lib/partial/transmuxer.js]: I believe it was intentional that there is no coalesceStream for the partial transmuxer in order to reduce the time to return data

File: lib/mp4/transmuxer.js [hunk 2]
```diff
@@ -709,14 +708,21 @@ CoalesceStream = function(options, metadataStream) {
     // important information required for the construction of
     // the final segment
     this.pendingTracks.push(output.track);
-    this.pendingBoxes.push(output.boxes);
     this.pendingBytes += output.boxes.byteLength;
 
+    // TODO: is there an issue for this against chrome?
```
Comment (brandonocasey) [inline lib/mp4/transmuxer.js]: This is what caused the test change, and a lot of the headache I was seeing getting this to work

File: lib/mp4/transmuxer.js [hunk 3]
```diff
@@ -21,8 +21,8 @@ var AdtsStream = require('../codecs/adts.js');
 var H264Stream = require('../codecs/h264').H264Stream;
 var AacStream = require('../aac');
 var isLikelyAacData = require('../aac/utils').isLikelyAacData;
+var ONE_SECOND_IN_TS = require('../utils/clock').ONE_SECOND_IN_TS;
```
Comment (brandonocasey) [inline lib/mp4/transmuxer.js]: This isn't a required change but we use it this way in most other files. I think I missed this during the original rebase.

File: lib/partial/audio-segment-stream.js [hunk 4]
```diff
@@ -4,6 +4,7 @@ var Stream = require('../utils/stream.js');
 var mp4 = require('../mp4/mp4-generator.js');
 var audioFrameUtils = require('../mp4/audio-frame-utils');
 var trackInfo = require('../mp4/track-decode-info.js');
+var ONE_SECOND_IN_TS = require('../utils/clock').ONE_SECOND_IN_TS;
```
Comment (brandonocasey) [inline lib/partial/audio-segment-stream.js]: This isn't a required change but we use it this way in most other files. I think I missed this during the original rebase.

File: test/transmuxer.test.js [hunk 5]
```diff
@@ -3738,7 +3738,7 @@ validateTrackFragment = function(track, segment, metadata, type) {
 QUnit.test('parses an example mp2t file and generates combined media segments', function() {
   var
     segments = [],
-    i, j, boxes, mfhd, trackType = 'video', trackId = 256, baseOffset = 0, initSegment;
+    i, j, boxes, mfhd, trackType = 'audio', trackId = 257, baseOffset = 0, initSegment;
```
Comment (brandonocasey) [inline test/transmuxer.test.js]: Since we now always output video first, this test had to be changed.

File: lib/mp4/transmuxer.js [hunk 6]
```diff
@@ -1119,6 +1129,16 @@ Transmuxer = function(options) {
     }
   };
 
+  this.setRemux = function(val) {
```
Comment (brandonocasey) [inline lib/mp4/transmuxer.js]: This allows us to set the `remux` option after the transmuxer is created.

**Pull request merged**